src/core/src/node_input.cpp

```diff
@@ -61,11 +61,13 @@ bool Input<Node>::operator!=(const Input& other) const {
     return !(*this == other);
 }
 bool Input<Node>::operator<(const Input& other) const {
-    return m_node < other.m_node || (m_node == other.m_node && m_index < other.m_index);
```

Comment:
also for `operator == `

Reply:
Is it needed? if addresses are equal, than instance id also should be equal? Or not?

Reply:
In operators< and > I also don't use instance id for checking equivalence

Reply:
ok, agree